test/unit/static_registers_unit_test.rb

```diff
@@ -5,252 +5,154 @@
 class StaticRegistersUnitTest < Minitest::Test
   include Liquid
 
-  def set
-    static_register        = StaticRegisters.new
-    static_register[nil]   = true
-    static_register[1]     = :one
-    static_register[:one]  = "one"
-    static_register["two"] = "three"
-    static_register["two"] = 3
-    static_register[false] = nil
-
-    assert_equal({ nil => true, 1 => :one, :one => "one", "two" => 3, false => nil }, static_register.registers)
-
-    static_register
+  def test_set
+    static_register = StaticRegisters.new(a: 1, b: 2)
+    static_register[:b] = 22
+    static_register[:c] = 33
+
+    assert_equal(1, static_register[:a])
+    assert_equal(22, static_register[:b])
+    assert_equal(33, static_register[:c])
   end
 
   def test_get
-    static_register = set
-
-    assert_equal(true, static_register[nil])
-    assert_equal(:one, static_register[1])
-    assert_equal("one", static_register[:one])
-    assert_equal(3, static_register["two"])
-    assert_nil(static_register[false])
-    assert_nil(static_register["unknown"])
+    static_register = StaticRegisters.new(a: 1, b: 2)
+    static_register[:b] = 22
+    static_register[:c] = 33
+
+    assert_equal(1, static_register[:a])
+    assert_equal(22, static_register[:b])
+    assert_equal(33, static_register[:c])
+    assert_nil(static_register[:d])
   end
 
   def test_delete
-    static_register = set
+    static_register = StaticRegisters.new(a: 1, b: 2)
+    static_register[:b] = 22
+    static_register[:c] = 33
 
-    assert_equal(true, static_register.delete(nil))
-    assert_equal(:one, static_register.delete(1))
-    assert_equal("one", static_register.delete(:one))
-    assert_equal(3, static_register.delete("two"))
-    assert_nil(static_register.delete(false))
-    assert_nil(static_register.delete("unknown"))
+    assert_nil(static_register.delete(:a))
+    assert_equal(1, static_register[:a])
```

Comment:
Is this behaviour intentional?  It wasn't previously covered by `test_delete`.  If the registers are supposed to behave like a hash, then having delete not actually behave like a delete seems like unexpected behaviour.

Reply:
> Is this behaviour intentional?

My best guess is that it is not, but it is what the code does.

This also have implication over how `key?` behave.

I'm not sure how these methods were chosen vs some others that were removed when moving from `Hash` to `StaticRegisters` (eg.: `.each` or `empty?`).

This might be going back to "StaticRegisters" shouldn't exist in the first place.

Reply:
> This also have implication over how `key?` behave.

Yeah, to maintain the current API, we would either need to keep track of deleted keys separately or to have a deleted key marker (e.g. similar to the `UNDEFINED` constant).  That would mean we would always have to retrieve the value and special case this marker value.  That makes me wonder if it is worth supporting that complexity to preserve this approach.

> I'm not sure how these methods were chosen vs some others that were removed when moving from `Hash` to `StaticRegisters` (eg.: `.each` or `empty?`).

Good question.  Perhaps it means we actually rely on this `delete` method, but if we don't need it, it would be simpler to not support.

> This might be going back to "StaticRegisters" shouldn't exist in the first place.

Indeed.  This would be much simpler if we just had two separate hashes on the context where `context.registers` could be initialized with `context.static_registers.dup`.  If this lazy approach is a performance optimization, then it seems like it was added prematurely.

Reply:
For this PR, maybe we should just remove the assertions for the value after it is deleted, rather than making it seems like the current behaviour is intentional.

Reply:
Deal, I'm fine with ship and iterate.